### ISSUE TITLE
[TW-436] line wrapping, punc as final character.

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#471]
_t charles yun on 2013-02-04T14:10:58Z says:_ 

 I have read http://taskwarrior.org/issues/804 and believe that the below isa sub-set case, but mention it just in case it is worth mentioning.

It seems that if the last character is a punctuation character, line parsing can get weird.  I ran into this with my display script and have worked around it.  In trying to isolate the problem came up with the following test case.

Create a task with annotations.  One of the annotations has a final punctuation mark.  If the line with the punctuation mark is the same length as the longest line, then the punctuation line will wrap oddly.  It is odd because the line should have enough screen space to not wrap.  If you add a longer annotation, the oddly wrapped line will un-wrap.  So it seems that the final punctuation mark is an issue.

Should whatever the counter is that is accumulating the length of the line ignore the final punctuation mark?

<pre>
$ task punc
[task next punc]

ID  Project Pri Due A Age Urgency Description
* -- ------- --- --- - --- ------- -----------
440                   26s       0 punc test

1 task

$ task 440 ann "asd."
Annotating task 440 'punc test'.
Annotated 1 task.

$ task 440 ann "123"
Annotating task 440 'punc test'.
Annotated 1 task.

$ task 440
[task next 440]

ID  Project Pri Due A Age Urgency Description
* -- ------- --- --- - --- ------- -----------------
440                   41s     0.9 punc test
                                    2013/02/04
                                  asd.
                                    2013/02/04 123

1 task

</pre>

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2013-02-04T14:10:58Z
Modified: 2014-02-09T01:54:30Z
```

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2013-02-11T03:17:15Z says:_ 

 Thank you for the bug report.  I just fixed #1136 today, and I believe this is a duplicate lof that.  Regardless, it is fixed now (as shown below), and the problem was the word-wrapping was confused by the indented annotations.

<pre>
$ TASKDATA=foo task add punc test
TASKDATA override: foo
Created task 1.
$ TASKDATA=foo task
TASKDATA override: foo
[task next]

ID Age Urgency Description
 1 4s        0 punc test  

1 task
$ TASKDATA=foo task 1 ann 'asd.'
Annotating task 1 'punc test'.
Annotated 1 task.
TASKDATA override: foo
$ TASKDATA=foo task 1 ann '123'
Annotating task 1 'punc test'.
Annotated 1 task.
TASKDATA override: foo
$ TASKDATA=foo task
TASKDATA override: foo
[task next]

ID Age Urgency Description      
 1 44s     0.9 punc test        
                 2/10/2013 asd. 
                 2/10/2013 123  

1 task
$ TASKDATA=foo task list
TASKDATA override: foo

ID Age Description      
 1 53s punc test        
         2/10/2013 asd. 
         2/10/2013 123  

1 task
$ TASKDATA=foo task long
TASKDATA override: foo

ID Added     Recur Age Description      
 1 2/10/2013     - 1m  punc test        
                         2/10/2013 asd. 
                         2/10/2013 123  

1 task
</pre>
